tests/test_grant_tagger.py

```diff
@@ -137,13 +130,30 @@ def test_train_test_info():
         prediction_cols=prediction_cols,
         label_name=label_name,
         )
-
-    X_vect, y = grant_tagger.fit_transform(training_data, train_data_id)
-    X_train, X_test, y_train, y_test = grant_tagger.split_data(X_vect, y)
+    X_train, X_test, y_train, y_test = grant_tagger.split_data(training_data, train_data_id)
```

Comment:
what is the point of having a split_data function in the class? is this function the responsibility of the class? also what does it promise to the user?

split_data sounds like a function that should live outside the class

Reply:
It's useful to keep it in the class to keep a record of which IDs were in the train/test/unseen split. But its true this function could return everything, i.e.
```
def split_data(data, train_data_id="Internal ID", label_name='Relevance code', split_seed=42, relevant_sample_ratio=1):
...
return X_train, X_test, X_unused, y_train, y_test, y_unused, X_train_ids, X_test_ids, unused_ids
```

Reply:
note that this can be a source of problems as I mentioned in that it returns a list of data and keeps an order with the ids, hoping that the user will not shuffle or split in any way